### PR TITLE
fixes #537: Kafka sink connector does not work with CUD ingestion strategy and Avro format data

### LINKED ISSUE #537: Kafka sink connector does not work with CUD ingestion strategy and Avro format data
Hello,
I am trying to use the sink connector to read messages from kafka topic (topic use AVRO format and schema registry for key and value) and sending to neo4j using CUD strategy, but can not get it to work.
If  the message in the kafka topic uses JSON format, it works correctly.
Also used the Neo4j streams plugin (sink using CUD strategy) and that works correctly for the same AVRO data.
It looks to me that the specific combination of sink connector/AVRO data is not working as expected when using CUD strategy.

## Expected Behavior (Mandatory)
Sink connector should work with AVRO format data when using CUD strategy

## Actual Behavior (Mandatory)
Sink connector fails to write the data to neo4j database, does not produce any errors in the connect log.

## How to Reproduce the Problem
Write CUD format data to topic in AVRO format - Example of the data I used to test with -
{"keycol":"26","type":"node","labels":["EMPLOYEE"],"op":"merge","ids":{"ID_NUM":"36950"},"properties":{"ID_NUM":"36950","EMP_NAME":"Edythe","PHONE":"3333"}}

Set up Kafka Neo4j sink connector and start connector. Connector will run but not produce any updates to neo4j.
Connect log shows -
```
[2022-09-12 18:10:25,937] TRACE WorkerSinkTask{id=neo4jcudsink-0} Polling returned 1 messages (org.apache.kafka.connect.runtime.WorkerSinkTask)
[2022-09-12 18:10:25,937] TRACE WorkerSinkTask{id=neo4jcudsink-0} Consuming and converting message in topic 'TONEO4J' partition 0 at offset 0 and timestamp 1663006471925 (org.apache.kafka.connect.runtime.WorkerSinkTask)
[2022-09-12 18:10:26,141] TRACE WorkerSinkTask{id=neo4jcudsink-0} Applying transformations to record in topic 'TONEO4J' partition 0 at offset 0 and timestamp 1663006471925 with key 26 and value Struct{type=node,labels=[EMPLOYEE],op=merge,ids=Struct{ID_NUM=36950},properties=Struct{ID_NUM=36950,EMP_NAME=Edythe,PHONE=3333}} (org.apache.kafka.connect.runtime.WorkerSinkTask)
[2022-09-12 18:10:26,144] TRACE WorkerSinkTask{id=neo4jcudsink-0} Delivering batch of 1 messages to task (org.apache.kafka.connect.runtime.WorkerSinkTask)
```

The sink task is reading the data successfully from the topic, is failing during processing.

Currently used versions

### Versions

  - Neo4j: 4.4.11
  - Neo4j Connector : neo4j/kafka-connect-neo4j:2.0.2

Any thoughts/ideas on this? Would like to confirm if it is a bug or my data has a problem.

Regards,
Vijay

### PR DESCRIPTION
**Pull request by conker84**
Fixes #537

The bug was related to the fact that the strategy expects a Map<String, Any?> (with the raw values) instead it gets a Map<String, Value>. I changed the converter in order to return Map<String, Any?> (with the raw values)

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - fixed the bug 
  - implemented test

### CODE REVIEW

File: kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jSinkTaskTest.kt [hunk 1]
```diff
@@ -1351,6 +1353,54 @@ class Neo4jSinkTaskTest {
         }
     }
 
+    @Test
+    fun `should successfully parse nested timestamps`() {
```
Comment (ali-ince) [inline kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jSinkTaskTest.kt:1357]: is this just timestamps, or more general sub-maps?

File: kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jValueConverterTest.kt [hunk 2]
```diff
@@ -174,11 +179,55 @@ class Neo4jValueConverterTest {
...
-        assertEquals(date.toInstant().atZone(ZoneId.of("UTC")).toLocalDateTime(), result["date"]?.asLocalDateTime())
+        assertEquals(double, result["double"])
+        assertEquals(long, result["long"])
+        assertEquals(bigDouble.toPlainString(), result["bigDouble"])
+        assertEquals(string, result["string"])
+        assertEquals(date.toInstant().atZone(ZoneId.of("UTC")).toLocalDateTime(), result["date"])
+    }
+
+    @Test
+    fun `should be able to process a CUD format AVRO structure`() {
```
Comment (ali-ince) [inline kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jValueConverterTest.kt:190]: should we also include a test for relationship CUD format?

**Review by ali-ince — COMMENTED** [review]
Looks good, just a couple of minor things.

File: kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jSinkTaskTest.kt (hunk 1 above)
Comment (conker84) [reply kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jSinkTaskTest.kt:1357]: is general sub-maps

File: kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jValueConverterTest.kt (hunk 2 above)
Comment (conker84) [reply kafka-connect-neo4j/src/test/kotlin/streams/kafka/connect/sink/Neo4jValueConverterTest.kt]: done

**Review by ali-ince — APPROVED** [review]
LGTM

**Pull request merged**